provider/datagen/src/lib.rs

```diff
@@ -435,52 +435,6 @@ fn keys_from_bin_inner(bytes: &[u8]) -> Vec<DataKey> {
     result
 }
 
-use icu_locale_core::langid;
-
-#[cfg(feature = "provider")]
-fn lstm_model_name_to_data_locale(name: &str) -> Option<DataLocale> {
```

Comment:
Thought: It's nice that we can get rid of these functions. I suggested remapping with new names, but maybe the new names should be applied deep inside DatagenProvider such that users of DatagenDriver also use the new names.